doc/stages/writers.gdal.rst

```diff
@@ -0,0 +1,123 @@
+.. _writers.gdal:
+
+writers.gdal
+================================================================================
+
+The `GDAL`_ writer creates a raster from a point cloud using an interpolation
+algorithm.  Output is produced using GDAL and can therefore use any `driver
+that supports creation of rasters`_.
+
+.. _`GDAL`: http://gdal.org
+.. _`driver that supports creation of rasters`: http://www.gdal.org/formats_list.html
+
+The technique used to create the raster is a simple interpolation where
+each point that falls within a given radius of a raster cell center
+potentially contributes to the raster's value.
+
+.. note::
+
+    If a circle with the provided radius doesn't encompass the entire cell,
+    it is possible that some points will not be considered at all, including
+    those that may be within the bounds of the raster cell.
+
+The GDAL writer creates rasters using the data in the 'Z' dimension of the 
+points.  In a :ref:`pipeline_command` one can precede the GDAL writer with
+the ferry filter (:ref:`filters.ferry`) in order to copy data from some
+other dimension to the 'Z' dimension if desired.
+
+The writer will creates up to six rasters based on different statistics in
+the output dataset.  The order of the layers in the dataset is as follows:
+
+min
+    Give the cell the minimum value of all points within the given radius.
+
+max
+    Give the cell the maximum value of all points within the given radius.
+
+mean
+    Give the cell the mean value of all points within the given radius.
+
+idw
+    Cells are assigned a value based on `Shepard's inverse distance weighting`_
+    algorithm, considering all points within the given radius.
+
+count
+    Give the cell the number of points that lie within the given radius.
+
+stdev
+    Give the cell the population standard deviation of the points that lie
+    within the given radius.
+
+.. _`Shepard's inverse distance weighting`: https://en.wikipedia.org/wiki/Inverse_distance_weighting
+
+If no points fall within the circle about a raster cell, a secondary
+algorithm can be used to attempt to provide a value after the standard
+interpolation is complete.  If the window_size_ option is set to a non-zero
+value, a square of rasters surrounding an empty cell, and the value of each
+non-empty surrounding is averaged using inverse distance weighting to provide
+a value for the subject cell.  The value provided for window_size is the
+maximum horizontal or vertical distance that a donor cell may be in order to
+contribute to the subject cell (A window_size of 1 essentially creates a 3x3
+array around the subject cell.  A window_size of 2 creates a 5x5 array, and
+so on.)
+
+Cells that have no value after interpolation are given the empty value of -9999.
+
```

Comment:
Maybe a down-the-road feature, but GDALBands have NoData masking capability with their "mask bands". If `writers.gdal` were to provide a mask band, GDAL can do the right thing for NODATA for each driver type. Then we can finally default to something like -9999.